tests/ConfigTest.php

```diff
@@ -13,7 +16,21 @@ class ConfigTest extends TestCase
      */
     public function testItRetrievesConfigurations(string $key, mixed $expected)
     {
-        $config = new Config(__DIR__ . '/config');
+        /** @var ConfigProvider&MockInterface */
```

Comment:
This typehint is missing the variable name. The proper hint would be `/** @var ConfigProvider&MockInterface $configProvider */`, but even that would be wrong in this context. It's also inferred anyway, so this can be removed.